org.eclipse.xtext.xtext.generator/xtend-gen/org/eclipse/xtext/xtext/generator/ui/contentAssist/ContentAssistFragment2.java

```diff
@@ -241,6 +241,17 @@ protected GeneratedJavaFileAccess generateGenJavaProposalProvider() {
         return _xblockexpression_1;
       };
       final ArrayList<AbstractRule> remainingRules = IterableExtensions.<AbstractRule, ArrayList<AbstractRule>>fold(this.getGrammar().getRules(), CollectionLiterals.<AbstractRule>newArrayList(), _function_1);
+      _xblockexpression = this.generateGenJavaProposalProvider(assignments, remainingRules);
+    }
+    return _xblockexpression;
+  }
+  
+  /**
+   * @since 2.23
+   */
+  protected GeneratedJavaFileAccess generateGenJavaProposalProvider(final List<Assignment> assignments, final List<AbstractRule> remainingRules) {
```

Comment:
the current api here really is bad and i did not want to break. what do you think?